src/core/main.c

```diff
@@ -1182,8 +1184,15 @@ static int bump_rlimit_nofile(struct rlimit *saved_rlimit) {
                 arg_default_rlimit[RLIMIT_NOFILE] = rl;
         }
 
+        /* Get current RLIMIT_NOFILE maximum compiled into the kernel. */
+        r = read_one_line_file("/proc/sys/fs/nr_open", &nr_open);
+        if (r == 0)
+                r = safe_atollu(nr_open, &min_max);
```

Comment:
Thanks, done. I wonder whether we should also instead of using `64 * 1024` as fallback use the normally hard-coded `1024 * 1024`.